messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryUpdateEmitter.java

```diff
@@ -105,13 +109,17 @@ public boolean queryUpdateHandlerRegistered(SubscriptionQueryMessage<?, ?, ?> qu
         FluxSink<SubscriptionQueryUpdateMessage<U>> sink = processor.sink(backpressure.getOverflowStrategy());
         sink.onDispose(() -> updateHandlers.remove(query));
         FluxSinkWrapper<SubscriptionQueryUpdateMessage<U>> fluxSinkWrapper = new FluxSinkWrapper<>(sink);
-        updateHandlers.put(query, fluxSinkWrapper);
-
         Registration registration = () -> {
             updateHandlers.remove(query);
+            registrations.remove(query);
             return true;
         };
 
+        updateHandlers.put(query, fluxSinkWrapper);
+        registrations.put(query,registration);
+
+
```

Comment:
Small fry, but check the indentation of the new code and remove unnecessary empty lines.